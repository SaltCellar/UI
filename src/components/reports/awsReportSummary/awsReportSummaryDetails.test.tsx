import { shallow } from 'enzyme';
import React from 'react';
import {
  AwsReportSummaryDetails,
  AwsReportSummaryDetailsProps,
} from './awsReportSummaryDetails';

const props: AwsReportSummaryDetailsProps = {
  report: { data: [], total: { cost: { value: 100, units: 'USD' } } },
  label: 'label',
  formatValue: jest.fn(() => 'formatedValue'),
};

test('report total is formated', () => {
  const view = shallow(<AwsReportSummaryDetails {...props} />);
  expect(props.formatValue).toBeCalledWith(
    props.report.total.cost.value,
    props.report.total.cost.units,
    props.formatOptions
  );
  expect(view).toMatchSnapshot();
});

test('defaults value if report is not present', () => {
  const view = shallow(<AwsReportSummaryDetails {...props} report={null} />);
  expect(props.formatValue).not.toBeCalled();
  expect(view).toMatchSnapshot();
});

test('defaults value if report.total is not present', () => {
  const view = shallow(
    <AwsReportSummaryDetails
      {...props}
      report={{ ...props.report, total: null }}
    />
  );
  expect(props.formatValue).not.toBeCalled();
  expect(view).toMatchSnapshot();
});
