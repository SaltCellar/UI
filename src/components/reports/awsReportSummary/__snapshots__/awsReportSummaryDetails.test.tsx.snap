// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaults value if report is not present 1`] = `
<div
  className="css-e13jst"
>
  <div
    className="css-1w8hocc"
  >
    ----
  </div>
  <div
    className="css-1640ajm"
  >
    <div>
      label
    </div>
  </div>
</div>
`;

exports[`defaults value if report.meta is not present 1`] = `
<div
  className="css-e13jst"
>
  <div
    className="css-1w8hocc"
  >
    ----
  </div>
  <div
    className="css-1640ajm"
  >
    <div>
      label
    </div>
  </div>
</div>
`;

exports[`report total is formated 1`] = `
<div
  className="css-e13jst"
>
  <div
    className="css-1w8hocc"
  >
    formatedValue
  </div>
  <div
    className="css-1640ajm"
  >
    <div>
      label
    </div>
  </div>
</div>
`;
